desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py

```diff
@@ -251,9 +252,25 @@ def execute(self, notebook, snippet):
         }
       )
 
-    result = connection.execute(statement)
+    result = connection.execute(statement, async_=True)
```

Comment:
Right now, instead of hardcoding it for all, could we set async to True only if the dialect is hive?
(if not will probably break all the dialects)